### PR TITLE
Review request: Fourth milestone for react calculator app

### PR DESCRIPTION
**Pull request by chrystalme**
- I added a logic folder
- Created operate.js and calculate.js file
- In the operate.js file, I added logic for checking the operator and provided the logic.
- I implemented operate method on numbers in the calculate.js. Also, logic for the different buttonNames

### CODE REVIEW

File: src/logic/operate.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+      result = x / y;
+      break;
+    case 'x':
+      result = x * y;
+      break;
+    case '%':
+      result = x / 100;
+      break;
+    default:
+      return;
```
Comment (rindrajosia) [inline src/logic/operate.js:24]: Please make sure that you use functions like plus, minus, times, div, and modifier provided by Big.js

File: src/logic/calculate.js [hunk 2]
```diff
@@ -1,7 +1,59 @@
...
+  if (buttonName === '+/-') {
+    if (total && !next) {
+      result.total = operate(total, -1, 'x');
+    }
+    if (next) {
+      result.total = operate(next, -1, 'x');
+    }
+  }
 
+  if (operators.includes(buttonName) /* && buttonName === operation */) {
```
Comment (rindrajosia) [inline src/logic/calculate.js:36]: Inline comments: It’s better to delete commented code. If you need to review a previous version of your code, use git history.

File: src/components/app.js [hunk 3]
```diff
@@ -1,9 +1,11 @@
 import React from 'react';
 import Display from './display';
 import ButtonPanel from './buttonPanel';
+import calculate from '../logic/calculate';
 import '../App.css';
 
 function App() {
```
Comment (rindrajosia) [inline src/components/app.js:7]: please use the arrow function instead of the function keyword.

**Review by rindrajosia — CHANGES_REQUESTED** [review]
Changes Required :red_circle:
To Highlight: :angel:

- calculate.js module is added :+1:
- operate.js module is added :+1:

To improve :factory_worker: [REQUIRED]:

- [ ] I got an error when I open your APP on my browser.


- [ ] It seems that you forgot to implement `equal operator ( = )` for buttonName in your calculate.js file.
- [ ] Inline comments: It’s better to delete commented code. If you need to review a previous version of your code, use git history.
- [ ] Please make sure that you use functions like plus, minus, times, div, and modifier  provided by Big.js 
([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/calculator/milestone4/performOperate.png))

- [ ] As you are using ES6 syntax, please use the arrow function instead of the function keyword.

Nice to have [OPTIONAL] :

- [ ] 

That's it, you can submit another code review request after making these changes :recycle: :innocent:
With all my respect :pray:
Rindra Josia

**Comment by chrystalme** [comment]
- I Removed the comment on calculate.js
 - Declared the variables in the calculatorData object before destructuring
 - Included logic for '='

Thanks for the review

**Review by afanimartin — APPROVED** [review]
## Approved 👍🏿 👍🏿 👍🏿 
Nice work making the requested changes.
### Highlights 🌵 🌵 🌵 
- comments removed
- use of `Big.js` for operations
- app runs locally without errors

### Martin Afani
#### Slack: whiz25

**Pull request merged**